api/handlers/service_binding_test.go

```diff
@@ -72,476 +72,555 @@ var _ = Describe("ServiceBinding", func() {
+				Expect(message.LabelSelector).To(Equal("label=value"))
+				Expect(message.PlanGUIDs).To(ConsistOf("p1", "p2"))
 
-		When("decoding URL params fails", func() {
-			BeforeEach(func() {
-				requestValidator.DecodeAndValidateURLValuesReturns(errors.New("boom"))
+				Expect(rr).To(HaveHTTPStatus(http.StatusOK))
+				Expect(rr).To(HaveHTTPHeaderWithValue("Content-Type", "application/json"))
+				Expect(rr).To(HaveHTTPBody(SatisfyAll(
+					MatchJSONPath("$.pagination.total_results", BeEquivalentTo(1)),
+					MatchJSONPath("$.pagination.first.href", "https://api.example.org/v3/service_credential_bindings?foo=bar"),
+					MatchJSONPath("$.resources[0].guid", "service-binding-guid"),
+				)))
 			})
 
-			It("returns an error", func() {
-				expectUnknownError()
+			When("there is an error fetching service binding", func() {
+				BeforeEach(func() {
+					serviceBindingRepo.ListServiceBindingsReturns([]repositories.ServiceBindingRecord{}, errors.New("unknown"))
+				})
+
+				It("returns an error", func() {
+					expectUnknownError()
+				})
 			})
-		})
-	})
 
-	Describe("DELETE /v3/service_credential_bindings/:guid", func() {
-		BeforeEach(func() {
-			requestMethod = "DELETE"
-			requestPath = "/v3/service_credential_bindings/service-binding-guid"
-		})
+			When("an include=app query parameter is specified", func() {
+				BeforeEach(func() {
+					payload := payloads.ServiceBindingList{
+						Include: "app",
+					}
+					requestValidator.DecodeAndValidateURLValuesStub = decodeAndValidateURLValuesStub(&payload)
+				})
 
-		It("gets the service binding", func() {
-			Expect(serviceBindingRepo.GetServiceBindingCallCount()).To(Equal(1))
-			_, actualAuthInfo, actualBindingGUID := serviceBindingRepo.GetServiceBindingArgsForCall(0)
-			Expect(actualAuthInfo).To(Equal(authInfo))
-			Expect(actualBindingGUID).To(Equal("service-binding-guid"))
-		})
+				It("includes app data in the response", func() {
+					Expect(appRepo.ListAppsCallCount()).To(Equal(1))
+					_, _, listAppsMessage := appRepo.ListAppsArgsForCall(0)
+					Expect(listAppsMessage.Guids).To(ContainElements("app-guid"))
 
-		When("getting the service binding is forbidden", func() {
-			BeforeEach(func() {
-				serviceBindingRepo.GetServiceBindingReturns(repositories.ServiceBindingRecord{}, apierrors.NewForbiddenError(nil, repositories.ServiceBindingResourceType))
+					Expect(rr).To(HaveHTTPBody(MatchJSONPath("$.included.apps[0].name", "some-app-name")))
+				})
 			})
 
-			It("returns a not found error", func() {
-				expectNotFoundError(repositories.ServiceBindingResourceType)
+			When("decoding URL params fails", func() {
+				BeforeEach(func() {
+					requestValidator.DecodeAndValidateURLValuesReturns(errors.New("boom"))
+				})
+
+				It("returns an error", func() {
+					expectUnknownError()
+				})
 			})
 		})
 
-		When("getting the service binding fails", func() {
+		Describe("DELETE /v3/service_credential_bindings/:guid", func() {
```

Comment:
`"DELETE /v3/service_credential_bindings/:guid"` is now under `Describe("POST /v3/service_credential_bindings"` which is not correct. It should be on the same level as `Describe("POST /v3/service_credential_bindings"`